app/models/manageiq/providers/vmware/cloud_manager/orchestration_service_option_converter.rb

```diff
@@ -20,60 +19,94 @@ def stack_create_options
-        src_vm = { :vm_id => "vm-#{vm_id}" }
-        src_vm[:name] = vm_opts["instance_name"] if vm_opts.key?("instance_name")
-
-        network_id = vm_opts["vdc_network"]
-        unless network_id.nil?
-          # Create new network config if it hasn't been created before.
-          network_config[network_id] ||= {
-            :networkName => network_id,
-            :networkId   => network_id,
-            :fenceMode   => "bridged"
+        src_vm = {
+          :vm_id    => "vm-#{vm_id}",
+          :networks => parse_nics(vm_opts),
+          :hardware => {
+            :cpu    => { :num_cores => vm_opts['num_cores'], :cores_per_socket => vm_opts['cores_per_socket'] },
+            :memory => { :quantity_mb => vm_opts['memory_mb'] },
+            :disk   => vm_opts['disk_mb']&.map { |disk| { :id => disk[:disk_id], :capacity_mb => disk[:value] } }
           }
-
-          # Add network configuration to the source VM.
-          src_vm[:networks] = [
-            :networkName             => network_id,
-            :IsConnected             => true,
-            :IpAddressAllocationMode => "DHCP"
-          ]
-        end
-
+        }
+        src_vm[:name] = vm_opts["instance_name"] if vm_opts.key?("instance_name")
+        src_vm[:guest_customization] = { :ComputerName => vm_opts['hostname'] } if vm_opts.key?("hostname")
         src_vm
       end
 
-      # Create options suitable for VMware vCloud provider.
-      custom_opts = {
-        :source_vms => source_vms
-      }
-      custom_opts[:InstantiationParams] = {
-        :NetworkConfig => network_config.values
-      } unless network_config.empty?
+      vapp_networks = vapp_net_params.collect do |vapp_net_name, opts|
+        {
+          :name       => vapp_net_name,
+          :parent     => opts['parent'],
+          :fence_mode => opts['fence_mode'],
+          :subnet     => parse_subnets(opts)
+        }
+      end
 
-      custom_opts
+      {
+        :source_vms    => source_vms,
+        :vapp_networks => vapp_networks
+      }
     end
 
     def collect_vm_params
-      allowed_vm_params = %w(instance_name vdc_network)
-      stack_parameters.each_with_object({}) do |(key, value), vm_params|
-        allowed_vm_params.each do |param|
-          # VM-specific parameters are named as instance_name-<VM_ID>. The
-          # following will test the param name for this kind of pattern and use
-          # the <VM_ID> to store the configuration about this VM.
-          param_match = key.match(/#{param}-([0-9a-f-]*)/)
+      collect_stack_parameters(
+        %w(instance_name vdc_network num_cores cores_per_socket memory_mb disk_mb hostname nic_network nic_mode nic_ip_address)
+      )
+    end
+
+    def collect_vapp_net_params
+      collect_stack_parameters(%w(gateway netmask dns1 dns2 parent fence_mode))
+    end
+
+    def collect_stack_parameters(allowed)
+      separator = '___'
+      stack_parameters.each_with_object({}) do |(key, value), params|
```

Comment:
@miha-plesko you might have already but can you paste an example of what stack_parameters looks like?

I think we could do this with pre-parsing and hashes instead of regex, base64, and yaml but you might have already tried that

Reply:
Here goes the `stack_parameters`:

```ruby
{"deploy"=>"f",
 "powerOn"=>"f",
 "parent-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___"=>"b915be99-1471-4e51-bcde-da2da791b98f",
 "fence_mode-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___"=>"bridged",
 "gateway-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"192.168.43.1",
 "netmask-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"255.255.255.0",
 "dns1-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"192.168.43.1",
 "dns2-UmVkSGF0IFByaXZhdGUgbmV0d29yayA0Mw==___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>nil,
 "parent-RGVmYXVsdCBOZXR3b3Jr___"=>nil,
 "fence_mode-RGVmYXVsdCBOZXR3b3Jr___"=>"isolated",
 "gateway-RGVmYXVsdCBOZXR3b3Jr___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"192.168.2.1",
 "netmask-RGVmYXVsdCBOZXR3b3Jr___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"255.255.255.0",
 "dns1-RGVmYXVsdCBOZXR3b3Jr___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>"8.8.8.8",
 "dns2-RGVmYXVsdCBOZXR3b3Jr___LS0tCjppcF9zY29wZV9pZHg6IDAK"=>nil,
 "instance_name-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___"=>"Database VM",
 "hostname-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___"=>"DatabaseVM",
 "num_cores-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___"=>2,
 "cores_per_socket-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___"=>2,
 "memory_mb-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___"=>1024,
 "disk_mb-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___LS0tCjpkaXNrX2lkOiAnMjAwMCcK"=>4096,
 "nic_network-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___LS0tCjpuaWNfaWR4OiAnMCcK"=>"Default Network",
 "nic_mode-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___LS0tCjpuaWNfaWR4OiAnMCcK"=>"MANUAL",
 "nic_ip_address-M2ZiZWI4MDctMGEzMS00YzFmLWEwMDgtOWI4YzI5MDZkNjYw___LS0tCjpuaWNfaWR4OiAnMCcK"=>"192.168.2.100",
 "instance_name-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___"=>"Web Server VM",
 "hostname-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___"=>"WebServerVM",
 "num_cores-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___"=>4,
 "cores_per_socket-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___"=>4,
 "memory_mb-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___"=>2048,
 "disk_mb-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpkaXNrX2lkOiAnMjAwMCcK"=>4096,
 "nic_network-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMCcK"=>"RedHat Private network 43",
 "nic_mode-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMCcK"=>"DHCP",
 "nic_ip_address-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMCcK"=>nil,
 "nic_network-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMScK"=>"Default Network",
 "nic_mode-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMScK"=>"DHCP",
 "nic_ip_address-MjVlYmQ3NDctOTI5OS00MjUzLTk0YTItNzQ4Y2QxNjI2OWI3___LS0tCjpuaWNfaWR4OiAnMScK"=>nil}
```